packages/frontend/src/contexts/attester.js

```diff
@@ -0,0 +1,94 @@
+import { makeAutoObservable } from 'mobx'
+import { SERVER } from '../config'
+
+export default class Attester {
+  epochsByAttester = []
+  signUpsByAttester = []
+  signUpsByEpoch = {}
+  attestationsByAttester = []
+  attestationsByEpoch = {}
+  totalPosRep = 0
+  totalNegRep = 0
+  allRep = 0
+  ustByAttester = []
+  ustByEpoch = {}
+
+  constructor(state) {
+    makeAutoObservable(this)
+    this.state = state
+    if (typeof window !== 'undefined') {
+      //   this.load()
+    }
+  }
+
+  //   async load() {}
+
+  async loadEpochsByAttester(attesterId) {
+    const url = new URL(`api/attester/${attesterId}/epochs`, SERVER)
+    const data = await fetch(url.toString()).then((r) => r.json())
+    this.epochsByAttester = data
+  }
+
+  async loadSignUpsByAttester(attesterId) {
+    const url = new URL(`api/attester/${attesterId}/signups`, SERVER)
+    const data = await fetch(url.toString()).then((r) => r.json())
+    this.signUpsByAttester = data
+    this.mapSignUpsToEpoch(data)
+  }
+
+  async mapSignUpsToEpoch(signups) {
+    this.signUpsByEpoch = new Map()
+    for (let i = 0; i < signups.length; i++) {
```

Comment:
Another style thing, I find it easier to iterate like `for (const signup of signups)`, it avoids having to access `signups[i]`

Reply:
def, so much cleaner thx 👍